browser/ui/webui/brave_shields/shields_data_handler.cc

```diff
@@ -49,6 +48,22 @@ void ShieldsDataHandler::GetSiteBlockInfo(GetSiteBlockInfoCallback callback) {
   std::move(callback).Run(site_block_info_.Clone());
 }
 
+void ShieldsDataHandler::GetAdBlockMode(GetAdBlockModeCallback callback) {
+  auto* shields_data_ctrlr = GetActiveShieldsDataController();
+  if (!shields_data_ctrlr)
```

Comment:
How could this ever happen? I'm not opposed to leaving this check necessarily, but I think we should also DCHECK in this case because it shouldn't ever happen. Also this will never respond back to the caller which might hang something in the UI? @petemill ?

Reply:
there's an interesting method if we're concerned about the callback will never run. `mojo::WrapCallbackWithDefaultInvokeIfNotRun()`

Reply:
from the mojo team

> To answer the question of whether or not it will hang, it just means the JS promise in the WebUI won't ever get called back

Reply:
actually it will cause a crash if a mojo callback gets destroyed without being run

Reply:
just trying to think about this again i'm not sure why we need `WrapCallbackWithDefaultInvokeIfNotRun` - If webui calls the method and an active controller doesn't exist the callback gets destroyed internally and webui as a caller won't get a value back. It will not crash/hang the webui at all. 

In DCHECK builds, we'll hit a DCHECK if the callback is destroyed without ever being run while the Receiver is connected. A DCHECK would suffice here.

Reply:
Yes it's fine not to call the callback when we've got a DCHECK as that will crash before the callback crash happens (not Running a mojom callback will cause a crash about callback getting destroyed before it was run in my experience) - but relying on a DCHECK means you're absolutely sure it won't happen at the moment 100% of the time. It's there to stop someone changing something in the future which causes e.g. something to be `nullptr`.